src/libsystemd/sd-netlink/netlink-socket.c

```diff
@@ -198,16 +198,14 @@ static int socket_recv_message(int fd, void *buf, size_t buf_size, uint32_t *ret
         assert(peek || (buf && buf_size > 0));
 
         n = recvmsg_safe(fd, &msg, MSG_TRUNC | (peek ? MSG_PEEK : 0));
-        if (n < 0) {
-                if (n == -ENOBUFS)
-                        return log_debug_errno(n, "sd-netlink: kernel receive buffer overrun");
-                if (ERRNO_IS_TRANSIENT(n)) {
-                        if (ret_mcast_group)
-                                *ret_mcast_group = 0;
-                        return 0;
-                }
+        if (n == -ENOBUFS)
+                return log_debug_errno(n, "sd-netlink: kernel receive buffer overrun");
+        else if (ERRNO_IS_NEG_TRANSIENT(n)) {
+                if (ret_mcast_group)
+                        *ret_mcast_group = 0;
+                return 0;
+        } else if (n < 0)
```

Comment:
Unnecessary `else`s.

Reply:
Because without the `else`, there'd need to be one more newline, and then the nice property that each "case" starts with one line would be broken.